dnf-behave-tests/features/environment.py

```diff
@@ -208,6 +208,41 @@ def get_cmd(self, context):
 
         return result
 
+    def get_microdnf_cmd(self, context):
+        result = ["microdnf"]
+        result.append(self.assumeyes_option)
+
+        # installroot can't be set via context for safety reasons
+        if self.installroot and self.installroot != "/":
+            result.append("--installroot={0}".format(self.installroot))
+            result.append("--noplugins")
+            result.append("--config={0}".format(self._get("config")))
+            result.append("--setopt=reposdir={0}".format(os.path.join(self.installroot, "etc/yum.repos.d")))
+            result.append("--setopt=varsdir={0}".format(os.path.join(self.installroot, "etc/dnf/vars")))
+            result.append("--setopt=cachedir={0}".format(os.path.join(self.installroot, "var/cache/yum")))
```

Comment:
@lukash 
Microdnf does not know what the user wants to do without these arguments.
I didn't introduce new arguments. The same arguments are in DNF. 
Yes, there is an algorithm in DNF that calculates default values. I didn't implement it. We discussed dafault behavior and arguments some time ago in the team. We didn't made the final decision. Time was running and `--installroot` was required for a long time. It is a key feature for some users.
I thought about it. Implementing all of these options and forcing them to use with `--installroot` was the answer to deal the situation.
User can use microdnf with `--installroot` now.
In any case, even with DNF I have to use these parameters, because the default values often do not suit me.